### ISSUE TITLE
Prompt server password before connection.

### ISSUE THREAD

**Issue by ZhukV** [deployphp/deployer#298]
How i can configure stage with prompt password to server before connect?

Thank.

**Comment by elfet** [comment]
Now it's not possible due many reasons: security, parallel execution, ext.

But i think it will be available when implement events in Deployer.

**Comment by ZhukV** [comment]
Ok. I now write codes for ask password before connecting. 1-2 hour(s) and i create pull request.

**Comment by ZhukV** [comment]
Now, with use PasswordGetter system you can customize logic for gets password.
In get password from `Configuration`, first checks `PasswordGetterInterface` instances, and if this it, then calls `getPassword($host, $user);`

As example, with use base callable (`CallablePasswordGetter`):

```php
use Deployer\Server\Password\CallablePasswordGetter;

server('production', 'domain.com')
    ->user('user')
    ->password(new CallablePasswordGetter(function ($host, $user){
        return 'some_password';
    }))
    ->stage('production')
    ->env('deploy_path', '/var/www/domain.com');
```

And with ask password (`AskPasswordGetter`):

```php
use Deployer\Server\Password\AskPasswordGetter;

server('production', 'domain.com')
    ->user('user')
    ->password(AskPasswordGetter::createLazyGetter())
    ->stage('production')
    ->env('deploy_path', '/var/www/domain.com');
```

**Attention:** in configure system, you must use lazy getter, because we should have a access to context for get ouput and input interfaces.

And, if send `null` password, then `AskPasswordGetter` will be created automatically.

```php
server('production', 'domain.com')
    ->user('user')
    ->password()
    ->stage('production')
    ->env('deploy_path', '/var/www/domain.com');
```

**Attention:** this method not working in parallel... I used this feature only on `SeriesExecutor`.

Thank.

**Comment by ZhukV** [comment]
This feature was released in #299
